requests/utils.py

```diff
@@ -816,6 +816,25 @@ def get_proxy(key):
     return False
 
 
+def getproxies_fix_system_proxies_windows():
+    """Fix for urllib getproxies() bug on Windows with Python < 3.10.5
+    See: https://bugs.python.org/issue42627
```

Comment:
Nothing precludes this from running on the versions unaffected according to you. This comment is aside from whether this is the correct fix

Reply:
This is a no-op for a fixed Python/urllib version. This commit checks whether getproxies() returns the same proxy host and port for three protocols, each prefixed with the protocol, and rewrites it to "http://". The fixed urllib already returns proxies with "http://", so this commit does nothing in this case.